File: Solutions/Python/Leaders_in_an_array.py
class Solution:
    #Back-end complete function Template for Python 3  
    #Function to find the leaders in the array.
    def leaders(self, A, N):
        #Code here
        Max = 0
        leaders = []
        for item in reversed(A):
            if item >= Max:
                leaders.append(item)
                Max = item
        return reversed(leaders)

File: Solutions/Java/Leaders_in_an_array.java
class Solution{
    //Function to find the leaders in the array.
    static ArrayList<Integer> leaders(int arr[], int n){
        // Your code here
        ArrayList<Integer> ans = new ArrayList<>();
        ans.add(arr[n-1]);       
        int leader = arr[n-1];       
        for (int i=n-2; i>=0; i--) {
            if (arr[i] >= leader) {
                ans.add(arr[i]);
                leader = arr[i];
            }
        }      
        Collections.reverse(ans);
        return ans;
    }
}
